app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,51 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :find_post
+  before_action :find_comment, only: [:edit, :update, :destroy]
+
+  def create
+    @comment        = Comment.new comment_params
+    @comment.post   = @post
+    @comment.user   = current_user
+
+    if @comment.save
+      CommentsMailer.notify_post_owner(@comment).deliver_later
+      redirect_to post_path(@post), notice: "Comment created."
+    else
+      flash[:alert] = "Comment was not updated."
+      render "/posts/show"
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @comment.update comment_params
```

Comment:
Cute lil protip™ (Gabe):

This comment can updated by anyone because we're not scoping the comment to the `current_user`. Someone could submit a form to update this comment and that user may not necessarily be the one who created it! Better instead to say

```ruby
@comment = current_user.comment.find(params[:id])
```